client/my-sites/themes/theme-showcase.jsx

```diff
@@ -600,6 +621,29 @@ class ThemeShowcase extends Component {
 					isSiteWooExpressOrEcomFreeTrial={ isSiteWooExpressOrEcomFreeTrial }
 					isSiteECommerceFreeTrial={ isSiteECommerceFreeTrial }
 				/>
+				<ThemeSiteSelectorModal
+					isOpen={ isSiteSelectorModalVisible }
+					navigateOnClose={ false }
+					onClose={ ( args ) => {
+						if ( args?.siteSlug ) {
+							this.redirectToSiteAssembler( { slug: args.siteSlug } );
+						}
+
+						this.setState( { isSiteSelectorModalVisible: false } );
+					} }
+				/>
+				<ThemeDesignYourOwnModal
+					isOpen={ isDesignThemeModalVisible }
+					onClose={ () => {
+						this.setState( { isDesignThemeModalVisible: false } );
+					} }
+					onCreateNewSite={ () => {
+						this.redirectToSiteAssembler();
+					} }
+					onSelectSite={ () => {
+						this.setState( { isDesignThemeModalVisible: false, isSiteSelectorModalVisible: true } );
+					} }
+				/>
```

Comment:
I think this approach is a bit confusing to read, as it's unclear that the Site Selector modal does not work "standalone", but only as a consequence of the Design Your Own one.

What do you think of wrapping them in a separate method, clarifying that they are supposed to only work together?

I'm thinking something very simple like this:

```jsx
renderAssemblerSiteSelectorModal = () => {
  /* get all needed props state etc. */
  return ( <>
    <ThemeSiteSelectorModal />
    <ThemeDesignYourOwnModal />
  </> );
}

// and in the main render function:

{ this.renderAssemblerSiteSelectorModal() }
```

Reply:
Thanks for the suggestion, addressed on https://github.com/Automattic/wp-calypso/pull/87660/commits/a2f604511de3e9278b15c29d3ea9a994ba607b85